libs/fs/src/list_directory.ts

```diff
@@ -117,15 +124,5 @@ export async function* listDirectory(
 export async function* listDirectoryRecursive(
   path: string
 ): AsyncGenerator<Result<FileSystemEntry, ListDirectoryError>> {
-  for await (const result of listDirectory(path)) {
-    if (result.isErr()) {
-      yield result;
-    } else {
-      const fileEntry = result.ok();
-      if (fileEntry.type === FileSystemEntryType.Directory) {
-        yield* listDirectoryRecursive(fileEntry.path);
-      }
-      yield result;
-    }
-  }
+  yield* listDirectory(path, Infinity);
```

Comment:
I kind of feel like it would make more sense from an API perspective for `listDirectoryRecursive` to take a `depth` argument rather than `listDirectory`. Though I don't feel super strongly

Reply:
I did this so `listRecursive` could be called where it is currently called and not change those files. Since I don't feel strongly about it either, going to just leave it.